edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/RestartPolicyManager.cs

```diff
@@ -86,6 +86,14 @@ bool ShouldRestart(IRuntimeModule module)
                 return shouldRestart;
             }
 
+            // On Raspberry Pi, we experienced that a taskCanceledException is thrown during module creation;
```

Comment:
I am not sure I follow the logic - are you saying we are getting into a situation where a container is created but never restarted _though it should be_, based on its desired Status? And that is because of a TaskCancelledException? 
Is it because of wrong architecture or something like that is preventing the container from ever starting properly?

Reply:
Yes, a plan is created to create influxdb with create and start command; however a taskCanceledException happened during create command and start command doesn't execute.  Therefore influxdb container is in created status and never started.  The docker image is correct, I can just run "docker start influxdb" without any issue.  Again, this is only happened on Raspberry Pi since it rans slower than x64.

Reply:
Why wouldn't the next reconcile start the container?

Reply:
Currently we only consider to restart if the runtime status is backoff.  You can find the logic just right above the change in this PR in RestartPolicyManager line 73.  Based on RestartPolicyManager.ComputeModuleStatusFromRestartPolicy method, status will not be set to backoff if runtime status is equal to stopped (refer to line 40 in RestartPolicyManager).

Since current logic use backoff module status to control restart module; I think it is better to update ComputeModuleStatusFromRestartPolicy method to stopped module status; will add "status == ModuleStatus.Stopped" in line 40.  What do you think?

Reply:
No, this fix is actually good. But I don't think we need the never before started check... If the module desired state is Running and the module is Stopped, then it should be restarted, regardless how it got into that state.

Reply:
Nit: Suggest updating the comment to state the desire, than how it was found.
Something like - 
 If the RuntimeStatus is Stopped, then the module should be always restarted.

Reply:
Actually wait - thinking about this a little more, I think we need to change HealthRestartPlanner a little bit. 
Can we solve this problem by simply increasing the timeout? Then we can file a bug to look into the HealthRestartPlanner..

Reply:
Do you know where I can set the timeout?

Reply:
It would be a parameter to the HttpClient used in ModuleManagementHttpClient. Let's try setting it to something like 300, to see if the issue goes away. It will at least help us root cause the issue.